osu.Server.Queues.ScorePump/Queue/ImportHighScores.cs

```diff
@@ -413,6 +414,13 @@ private async Task<ScoreInfo> createReferenceScore(Ruleset ruleset, HighScore hi
                     scoreInfo.MaximumStatistics[HitResult.LegacyComboIncrease] = maxComboFromAttributes.Value - maxComboFromStatistics;
 #pragma warning restore CS0618
 
+                ScoreProcessor scoreProcessor = ruleset.CreateScoreProcessor();
+                int baseScore = scoreInfo.Statistics.Where(kvp => kvp.Key.AffectsAccuracy()).Sum(kvp => kvp.Value * Judgement.ToNumericResult(kvp.Key));
+                int maxBaseScore = scoreInfo.MaximumStatistics.Where(kvp => kvp.Key.AffectsAccuracy()).Sum(kvp => kvp.Value * Judgement.ToNumericResult(kvp.Key));
+
+                scoreInfo.TotalScore = (int)scoreProcessor.ComputeScore(ScoringMode.Standardised, scoreInfo);
+                scoreInfo.Accuracy = maxBaseScore == 0 ? 1 : baseScore / (double)maxBaseScore;
```

Comment:
I'm very likely to be missing something here, but does this apply to legacy/stable scores? And if so, wouldn't the accuracy change here because slider ticks/ends are not present in the statistics? This is with the assumption that slider ticks/ends affect accuracy in stable.

Reply:
> does this apply to legacy/stable scores?

This entire thing applies only to legacy/stable scores.

> wouldn't the accuracy change

Yes, the accuracy does change. However it's not exactly how you think of it. Slider ticks/ends do not affect accuracy in stable, and the values computed by both osu!stable and osu!web are derived solely from the statistics available in scores.

Some cases do differ, such as mania, where in stable/web GREATs/PERFECTs are treated as the same value, but in lazer they're not. For taiko, stable uses 300/150 whereas lazer uses 300/100.

This information is recoverable, because, as I said, it's done in [real time from score statistics](https://github.com/ppy/osu/blob/8c202ce141c3f4b742e401caf0b47bb694a528d5/osu.Game/Scoring/Legacy/LegacyScoreDecoder.cs#L138-L235).

Reply:
Right, thanks for the clarification.